docs/getting-started/configuration.md

```diff
@@ -0,0 +1,307 @@
+like in the [Getting Started guide](./index.md) and want to configure Backstage.
+
+At the end of this tutorial, you can expect:
+
+- Backstage to use a PostgreSQL database
+- You'll authenticate using one of the auth providers
+- The Backstage GitHub integration to be configured
+- You're able to use Software Templates
+
+### Prerequisites
+
+- Access to a Linux-based operating system, such as Linux, MacOS or
+  [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/)
+- An account with elevated rights to install prerequisites
+- If the database is not hosted on the same server as the Backstage app, the
+  PostgreSQL port needs to be accessible (the default is 5432 or 5433)
+
+### Install and configure PostgreSQL
+
+These instructions can be skipped if you already have a PostgreSQL server
+installed and created a schema and user. The example below is for Linux, but
+luckily there's detailed instructions on how to
+[install PostgreSQL](https://www.postgresql.org/download/) to help you get
+started.
+
+```shell
+sudo apt-get install postgresql
+```
+
+Test if your database is working:
+
+```shell
+sudo -u postgres psql
+```
+
+You should see a very welcoming message, like:
+
+```shell
+psql (12.9 (Ubuntu 12.9-0ubuntu0.20.04.1))
+Type "help" for help.
+
+postgres=#
+```
+
+For this tutorial we're going to use the existing postgres user. The next step
+is to set the password for this user:
+
+```shell
+postgres=# ALTER USER postgres PASSWORD 'secret';
+```
+
+That's enough database administration to get started. Type `\q`, followed by
+pressing the enter key. Then again type `exit` and press enter. Next, you need
+to install and configure the client.
+
+Stop Backstage, and go to the root directory of your freshly installed Backstage
+App. Use the following commands to start the PostgreSQL client installation:
+
+```shell
+# From your Backstage root directory
+cd packages/backend
+yarn add pg
+```
+
+Use your favorite editor to open `app-config.yaml` and add your PostgreSQL
+configuration. in the root directory of your Backstage app using the credentials
+from the previous steps.
+
+```diff
+backend:
+  database:
+-    client: sqlite3
+-    connection: ':memory:'
++    # config options: https://node-postgres.com/api/client
++    client: pg
++    connection:
++      host: ${POSTGRES_HOST}
++      port: ${POSTGRES_PORT}
++      user: ${POSTGRES_USER}
++      password: ${POSTGRES_PASSWORD}
++      # https://node-postgres.com/features/ssl
++      #ssl: require # see https://www.postgresql.org/docs/current/libpq-ssl.html Table 33.1. SSL Mode Descriptions (e.g. require)
++        #ca: # if you have a CA file and want to verify it you can uncomment this section
++        #$file: <file-path>/ca/server.crt
+```
+
+You'll use the connection details from the previous step. You can either set the
+`POSTGRES_` environment variables prior to launching Backstage, or remove the
+`${...}` values and set actual values in this configuration file.
```

Comment:
TIP: Do not add passwords to `app-config.yaml` as the file can be accidently committed. Use `app-config.local.yaml` or other gitignored files for secrets. (Equivalent to `.env` files).

Reply:
That's true, the getting-started shouldn't be the base for going anywhere official though. 
How to configure B. "for production" Is a much-asked-for-doc. Love to see this pop up there, or separate doc to link too.